src/nspawn/nspawn.c

```diff
@@ -2740,9 +2766,12 @@ static int outer_child(
         if (fd < 0)
                 return fd;
 
+        if (arg_private_network || arg_network_namespace_path)
+                use_netns = true;
+
         pid = raw_clone(SIGCHLD|CLONE_NEWNS|
                         arg_clone_ns_flags |
-                        (arg_private_network ? CLONE_NEWNET : 0) |
+                        (use_netns ? CLONE_NEWNET : 0) |
```

Comment:
actually, thinking about this: this patch will now create a new network namespace, and then immediately replace it by the specified one. Hence there's no reason to create the intermin namespace at all!